dvc/analytics.py

```diff
@@ -1,256 +1,124 @@
+        "dvc_version": __version__,
+        "is_binary": is_binary(),
+        "scm_class": scm_in_use(),
+        "system_info": system_info(),
+        "user_id": find_or_create_user_id(),
+    }
 
-    URL = "https://analytics.dvc.org"
-    TIMEOUT_POST = 5
-
-    USER_ID_FILE = "user_id"
-
-    PARAM_DVC_VERSION = "dvc_version"
-    PARAM_USER_ID = "user_id"
-    PARAM_SYSTEM_INFO = "system_info"
-
-    PARAM_OS = "os"
-
-    PARAM_WINDOWS_VERSION_MAJOR = "windows_version_major"
-    PARAM_WINDOWS_VERSION_MINOR = "windows_version_minor"
-    PARAM_WINDOWS_VERSION_BUILD = "windows_version_build"
-    PARAM_WINDOWS_VERSION_SERVICE_PACK = "windows_version_service_pack"
-
-    PARAM_MAC_VERSION = "mac_version"
-
-    PARAM_LINUX_DISTRO = "linux_distro"
-    PARAM_LINUX_DISTRO_VERSION = "linux_distro_version"
-    PARAM_LINUX_DISTRO_LIKE = "linux_distro_like"
-
-    PARAM_SCM_CLASS = "scm_class"
-    PARAM_IS_BINARY = "is_binary"
-    PARAM_CMD_CLASS = "cmd_class"
-    PARAM_CMD_RETURN_CODE = "cmd_return_code"
-
-    def __init__(self, info=None):
-        from dvc.config import Config
-        from dvc.lock import Lock
-
-        if info is None:
-            info = {}
-
-        self.info = info
-
-        cdir = Config.get_global_config_dir()
-        try:
-            os.makedirs(cdir)
-        except OSError as exc:
-            if exc.errno != errno.EEXIST:
-                raise
 
-        self.user_id_file = os.path.join(cdir, self.USER_ID_FILE)
-        self.user_id_file_lock = Lock(self.user_id_file + ".lock")
+def is_enabled():
+    if env2bool("DVC_TEST"):
+        return False
 
-    @staticmethod
-    def load(path):
-        """Loads analytics report from json file specified by path.
+    enabled = to_bool(
+        Config(validate=False)
+        .config.get(Config.SECTION_CORE, {})
+        .get(Config.SECTION_CORE_ANALYTICS, "true")
+    )
 
-        Args:
-            path (str): path to json file with analytics report.
-        """
-        with open(path, "r") as fobj:
-            analytics = Analytics(info=json.load(fobj))
-        os.unlink(path)
-        return analytics
+    logger.debug("Analytics is {}enabled.".format("" if enabled else "dis"))
 
-    def _write_user_id(self):
-        import uuid
+    return enabled
 
-        with open(self.user_id_file, "w+") as fobj:
-            user_id = str(uuid.uuid4())
-            info = {self.PARAM_USER_ID: user_id}
-            json.dump(info, fobj)
-            return user_id
 
-    def _read_user_id(self):
-        if not os.path.exists(self.user_id_file):
-            return None
+def send(path):
+    url = "https://analytics.dvc.org"
 
-        with open(self.user_id_file, "r") as fobj:
-            try:
-                info = json.load(fobj)
-            except ValueError as exc:
-                logger.debug("Failed to load user_id: {}".format(exc))
-                return None
-
-            return info[self.PARAM_USER_ID]
+    with open(str(path)) as fobj:
```

Comment:
```suggestion
    with open(fspath_py35(path)) as fobj:
```

Reply:
I don't get the difference? @efiop, should I change it in other places?

Reply:
@mroutis open accepts path-like objects starting from py36, so there is no need to `str()` it there. This corresponds to the way we handle this stuff in the rest of the codebase.

Reply:
got it, will have it in mind for further PRs :slightly_smiling_face: